django/utils/datetime_safe.py

```diff
@@ -8,12 +10,20 @@
 # '0010/08/02 was a Monday'
 
 import time as ttime
+import warnings
 from datetime import (
     date as real_date, datetime as real_datetime, time as real_time,
 )
 
+from django.utils.deprecation import RemovedInDjango50Warning
 from django.utils.regex_helper import _lazy_re_compile
 
+warnings.warn(
+    'The django.utils.datetime_safe module is deprecated.',
```

Comment:
I wonder if it's worth pointing to the alternative here. 🤔

Reply:
I thought about that but wasn't sure. `datetime_safe` is currently only used in three places where `.strftime()` is called immediately afterwards so the implementation isn't leaked. But that may not have always been the case - ticket-29600 removed other uses of it and there were a couple of other tickets raised about bugs with it in the past.

It is also only mentioned in the release notes for 1.4.9 and 1.5.5 in relation to a bug with `.combine()`. But there are no other mentions in documentation. I figured that deprecation, instead of immediate removal, made sense given its history.

If you'd still like me to mention the alternative, let me know. But if we do, we'd probably want to document the alternative too?

Reply:
I would leave only `The django.utils.datetime_safe module is deprecated.`. This a private API, we don't see to provide an alternative.